src/OSPSuite.Core/Domain/Quantity.cs

```diff
@@ -147,19 +147,29 @@ public double ValueInDisplayUnit
          set => Value = this.ConvertToBaseUnit(value);
       }
 
-      public virtual bool TryGetValueInDisplayUnit(out double result)
+      private (double value, bool success) tryGetValue()
       {
-         try
-         {
-            result = ValueInDisplayUnit;
-            return true;
-         }
-         catch (OSPSuiteException)
+         if (IsFixedValue || _cachedValueValid)
+            return (_cachedValue, success: true);
+
+         if (Formula == null)
+            return (double.NaN, success: false);
+
+         var (value, success) = Formula.TryCalculate(this);
+         if (success)
          {
-           //Formula could not be parsed. In that case, we save the value as double.NaN so that next time, the value will be NaN
-           result = double.NaN;
-           return false;
+            _cachedValue = value;
+            //Cached value is only valid if the Formula has updated it's references
+            _cachedValueValid = Formula.AreReferencesResolved;
```

Comment:
I forgot all the details about cached value, but: does it make sense to set _cachedValue if _cachedValueValid is false?
In this case _cachedValue will not be used anyway, right ?

Reply:
This is the same code as for Value {set;} . I did not want to change the logic here...You are probably right

Reply:
going to check this out

Reply:
I am going to leave the code as is. Otherwise we need to add some more if checks for nothing.